src/plugins/stake-plugins/stakePlugins.ts

```diff
@@ -3,20 +3,31 @@ import { makeTcSaversPlugin } from './thorchainSavers/tcSaversPlugin'
 import { StakePlugin } from './types'
 import { makeUniV2StakePlugin } from './uniswapV2/uniV2Plugin'
 
+// Return the memoized plugins and update them in the background for the next time this function is called
+let loadedPlugins: StakePlugin[]
+
 export const makeStakePlugins = async (): Promise<StakePlugin[]> => {
   const tcInitOptions = typeof ENV.THORCHAIN_INIT === 'object' ? ENV.THORCHAIN_INIT : {}
 
-  const results = await Promise.all([
+  const promises = [
     makeUniV2StakePlugin().catch(e => {
       console.warn(e.message)
     }),
     makeTcSaversPlugin({ initOptions: tcInitOptions }).catch(e => {
       console.warn(e.message)
     })
-  ])
-  const out: StakePlugin[] = []
+  ]
+
+  if (loadedPlugins != null) {
+    Promise.all(promises)
+    return loadedPlugins
+  }
+  const results = await Promise.all(promises)
+
+  loadedPlugins = []
   results.forEach(result => {
-    if (result != null) out.push(result)
+    if (result != null) loadedPlugins.push(result)
   })
```

Comment:
Is this the same as `loadedPlugins = results.filter(result => result != null)`?

Reply:
It is, I just struggled with the typing and ignored it. Fixed.